src/components/DayStreak.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from "react";
+import { Text } from "react-native";
+import styles from "../../styles";
+import AsyncStorage from '@react-native-community/async-storage';
+
+/*
+ * Day streak. Display how many consecutive days the user has
+ * meditated for.
+ */
+const DayStreak = ({ navigation }: any) => {
+  const [streak, setStreak] = useState("");
+
+  /*
+   * Get the current count and the last day the user meditated.
+   */
+  const getStreakData = async () => {
+    // Worry what happens here if no key is found, however testing
+    // suggests this is fine.
+    try {
+      // Update last streak count and last day meditated
+      return {
+        count: Number(await AsyncStorage.getItem('@streak_Count')),
+        lastDay: new Date(Number(await AsyncStorage.getItem('@last_Day_Year')),
+                          Number(await AsyncStorage.getItem('@last_Day_Month')),
+                          Number(await AsyncStorage.getItem('@last_Day_Date')))
+      }
+    } catch (e) {
+      // Error getting data...
+    }
+    return {
+      // We hit an error, just return null
+      count: null,
+      lastDay: null
+    }
+  }
+
+  /*
+   * Save the current count and today's date meditated.
+   */
+  const updateStreakData = async (count: any,
+                                  currentDay: Date) => {
```

Comment:
this doesn't need to take currentDay? It will always be `new Date()` ?

Reply:
hmm probably, but I would like to keep it general in case we every need to update streak data in another place